src/freenas/usr/local/lib/middlewared_truenas/plugins/enclosure.py

```diff
@@ -160,8 +163,8 @@ def _get_slot(self, slot_filter, enclosure_query=None):
 
         raise MatchNotFound()
 
-    def _get_slot_for_disk(self, disk):
-        return self._get_slot(lambda element: element["data"]["Device"] == disk)
+    def _get_slot_for_disk(self, disk, enclosure_info=None):
+        return self._get_slot(lambda element: element["data"]["Device"] == disk, enclosure_info)
```

Comment:
Shouldn't it be `enclosure_info=enclosure_info`?

Reply:
Yeah, I fixed